bigquery/google/cloud/bigquery/job.py

```diff
@@ -819,6 +819,35 @@ def to_api_repr(self):
         """
         return copy.deepcopy(self._properties)
 
+    def fill_from_default(self, default_job_config):
+        """Merge this job config with another one.
+        The other takes precedence with conflicting keys.
+        This is a naive one level merge.
+
+        :type default_job_config: google.cloud.bigquery.job._JobConfig
+        :param default_job_config:
+            The default job config that will be used to fill in self.
+
+        :rtype: :class:`google.cloud.bigquery.job._JobConfig`
+        :returns: A new job config.
+        """
+        if self._job_type != default_job_config._job_type:
+            raise TypeError(
+                "attempted to merge two incompatible job types: "
+                + repr(self._job_type) + ', '
+                + repr(default_job_config._job_type))
+
+        new_job_config = self.__class__()
+        new_job_config._properties = copy.deepcopy(self._properties)
+
+        new_job_properties = new_job_config._properties[self._job_type]
+        default_job_properties = default_job_config._properties[self._job_type]
+        for key in default_job_properties:
```

Comment:
You could simplify this (remove the loops) by starting with the default properties and then calling [update](https://docs.python.org/3/library/stdtypes.html#dict.update) to set those from the new job.

```
default_job_properties = copy.deepcopy(default_job_config._properties)
default_job_properties.update(self._properties)
new_job_config._properties = default_job_properties
return new_job_config
```

Reply:
It seems to me that we'll need two `update`s, one for the top layer (grabbing things like labels and whatever else), and another for the nested `_job_type` keys since `update` doesn't recurse.

```python
default_job_properties = copy.deepcopy(default_job_config._properties)
default_job_properties.update({ key: self._properties[key] for key in self._properties if key != self._job_type })
default_job_properties[self._job_type].update(self._properties[self._job_type])
new_job_config._properties = default_job_properties
return new_job_config
```